pkg/options/types_test.go

```diff
@@ -61,3 +61,33 @@ func TestCollectorSetSet(t *testing.T) {
 		}
 	}
 }
+
+func TestNamespaceListSet(t *testing.T) {
+	tests := []struct {
+		Desc   string
+		Value  string
+		Wanted NamespaceList
+	}{
+		{
+			Desc:   "empty namespacelist",
+			Value:  "",
+			Wanted: NamespaceList{},
+		},
+		{
+			Desc:  "normal namespacelist",
+			Value: "default, kube-system",
+			Wanted: NamespaceList([]string{
+				"default",
+				"kube-system",
+			}),
+		},
+	}
+
+	for _, test := range tests {
+		ns := &NamespaceList{}
+		gotError := ns.Set(test.Value)
+		if gotError != nil || !reflect.DeepEqual(*ns, test.Wanted) {
```

Comment:
I would recommend declaring a common method for comparing errors in unit tests. This come in handy in the future. Can we implement something like this? https://github.com/prometheus/prometheus/blob/master/util/testutil/error.go

Also, I prefer not relying on `reflect` if we have other ways.